pkg/asset/manifests/openstack/cluster.go

```diff
@@ -33,35 +33,44 @@ func GenerateClusterAssets(installConfig *installconfig.InstallConfig, clusterID
 			},
 		},
 		Spec: capo.OpenStackClusterSpec{
-			CloudName: cloudName,
-			IdentityRef: &capo.OpenStackIdentityReference{
-				Kind: "Secret",
-				Name: clusterID.InfraID + "-cloud-config",
+			IdentityRef: capo.OpenStackIdentityReference{
+				Name:      clusterID.InfraID + "-cloud-config",
+				CloudName: cloudName,
 			},
 			// We disable management of most networking resources since either
 			// we (the installer) will create them, or the user will have
 			// pre-created them as part of a "Bring Your Own Network (BYON)"
 			// configuration
-			ManagedSecurityGroups:      false,
+			ManagedSecurityGroups:      nil,
 			DisableAPIServerFloatingIP: true,
 			// TODO(stephenfin): update when we support dual-stack (there are
 			// potentially *two* IPs here)
-			APIServerFixedIP:  openstackInstallConfig.APIVIPs[0],
-			DNSNameservers:    openstackInstallConfig.ExternalDNS,
-			ExternalNetworkID: openstackInstallConfig.ExternalNetwork,
+			APIServerFixedIP: openstackInstallConfig.APIVIPs[0],
+			ExternalNetwork:  capo.NetworkFilter{ID: openstackInstallConfig.ExternalNetwork},
```

Comment:
Instead of specifying the `ID` in the `NetworkFilter`, we should specify `Name` as the externalNetwork is a name:
    `externalNetwork: provider`

Reply:
good catch, thank you for your detailed reviews :pray: